integrade/tests/api/v1/test_cloud_accounts.py

```diff
@@ -248,3 +289,73 @@ def test_negative_create_cloud_account_missing(field_to_delete):
         auth=auth
     )
     assert create_response.status_code == 400
+
+
+@pytest.mark.serial_only
+@pytest.mark.skipif(len(config.get_config()[
+    'aws_profiles']) < 1, reason='needs at least 1 aws profile')
+def test_cloudtrail_updated(
+        drop_account_data,
+        cloudtrails_and_buckets_to_delete):
+    """Ensure the cloudtrail is updated if a pre-existing one is found.
+
+    :id: d0174576-9b7c-48f7-8556-b560badf062d
+    :description: Ensure that at cloud account creation, if a pre-existing
+        cloudigrade cloudtrail is found, it is updated.
+    :steps: 1) Create a user and authenticate with their password
+        2) Create a cloudtrail with the same name that cloudigrade uses but
+            a different s3 bucket.
+        3) Create a cloud account using this same AWS account.
+        4) Assert that the cloudtrail is updated and now uses correct s3
+            bucket.
+    :expectedresults:
+        1) The server returns a 201 response with the information
+            of the created account.
+        2) The cloudtrail is updated to use the correct s3 bucket.
+    """
+    auth = get_auth()
+    cfg = config.get_config()
+    aws_profile = cfg['aws_profiles'][0]
+    profile_name = aws_profile['name']
+    acct_arn = aws_profile['arn']
+    client = api.Client(authenticate=False)
+    cloudtrail_client = aws_utils.aws_session(
+        profile_name).client('cloudtrail')
+
+    # create our own cloud trail with a different s3 bucket
+    bucket_name = aws_utils.create_bucket_for_cloudtrail(profile_name)
+    cloudtrail_client.create_trail(
+        Name=aws_profile['cloudtrail_name'],
+        S3BucketName=bucket_name)
+    # make sure we clean up our trail and bucket even if test fails
+    cloudtrails_and_buckets_to_delete.append(
+        (profile_name, aws_profile['cloudtrail_name'], bucket_name))
+    cloud_account = {
+        'account_arn': acct_arn,
+        'resourcetype': 'AwsAccount'
+    }
+    create_response = client.post(
+        urls.CLOUD_ACCOUNT,
+        payload=cloud_account,
+        auth=auth
+    )
+    assert create_response.status_code == 201
```

Comment:
Something that I found useful when debugging some API tests in the past was to have the response text as the assert message, that would show why it has failed.